galaxyproject/galaxy
# Add Vizarr visualization

Requires #19040 and #19097
xref #16862

Visualization for Zarr-like datasets based on https://github.com/hms-dbmi/vizarr

## For local Zarr datasets (directories uploaded as zip)

https://github.com/user-attachments/assets/75806c9e-6ce0-4eb5-8b9d-ea32cbbb1a45

## For remote (deferred) Zarr datasets (URI pointing to an S3 or remote location)


https://github.com/user-attachments/assets/4f342eff-2f18-4951-86b6-e60e460a273c




## TODO
- [x] Move the visualization to the new "pre-built visualizations" best practice
- [x] Fix ZarrDirectory datatype not being associated with the visualization ([all the other datatypes work](https://github.com/galaxyproject/galaxy/compare/dev...davelopez:add_vizarr_visualization?expand=1#diff-93c6e01827e0977bf358dc19da62dc8def73af7112de78ca9aa4ed769fd7b428R9-R12)...)

## How to test the changes?
  - Using local Zarr datasets
    - Upload a zarr.zip or ome_zarr.zip file with image data
    - Unpack the zip to a directory using the appropriate datatype converter
    - Check the Vizarr visualization 
  - Using remote Zarr datasets (public)
    - Go to a public Zarr repository, for example, https://www.openmicroscopy.org/2020/11/04/zarr-data.html
    - Copy any link to a zarr-based store and paste it into the Galaxy upload dialog
    - Make sure you set `defer dataset resolution` in the upload options before uploading
    - Once it is uploaded, check the Vizarr visualization for the deferred dataset


## License
